lib/Cake/Network/CakeResponse.php

```diff
@@ -854,7 +855,7 @@ public function sharable($public = null, $time = null) {
 		}
 
 		$this->maxAge($time);
-		if (!$time) {
+		if ($time === null) {
```

Comment:
What about 0 or false or other fasely values?
By changing this here you possibly introduce dangerous BC breaks for existing apps.
Better not changing functionality in a maintenance release major anymore where not absolutely necessary. Here it doesnt seem like it.

Reply:
> The expression $time of type integer|null is loosely compared to false; this is ambiguous if the integer can be 0. You might want to explicitly use === null instead.

The code intention was not very clear here. Perhaps `if (empty($time)) {` would be better.

Reply:
No, "if empty" is also bad (and functionally equal to what was before), see my other message.
`if (!$time)` is perfect to be kept.

Reply:
The intention of the check becomes ambiguous when an interger is converted to a boolean. Another option would be to cast to an integer `if ((int)$time === 0) {` or to set the default value to 0 instead of `null` what I think makes more sense as `$time` is supposed to be an integer: 
`public function sharable($public = null, $time = 0) {`.

`if (!$time) {` check is also inconsistent with strict check `if ($seconds !== null) {` done in `maxAge()` method that is called on line 857.

Reply:
If that is the case, then casting sure would be more BC here, people could rely on falsely values to also trigger the ! part.